### PR TITLE
[Clean up] Remove obsolete functionnalities

### PR DESCRIPTION
**Pull request by lpetre-ulb**
## Description

This PR removes a bunch of obsolete functionalities. That should help moving towards the new repository architecture while improving navigation in the code.

The following unused parts of code have been removed.

* Remove local database support (technically supersedes #321)
  * Drop MySQL dependency
  * Drop runtime Python dependency from the `cmsgemos` core
* Remove obsolete `gemsupervisor` configurations
* Removes `gemHwMonitor` which is not compatible at all with v3 electronics and is based on a XML system database
* Remove the XML DB system (v2 electronics only; superseded by `gemonlinedb`)
* Remove VFAT devices and managers from `gemhardware`
  * VFAT2 is obsolete
  * VFAT3 is empty
* Remove the AMC readout functionality (block deleted from firmware)

If desired, this large, but simple, PR can easily be split removal per removal. The commits are however clear and follow the structure: remove any reference to the functionnality -> delete the files -> clean-up the XML configuration files. 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Clean up

## Motivation and Context

Clean up in preparation for future developments.

## How Has This Been Tested?

All the `cmsgemos` xDAQ code still compile and the applications can be started. ~~State transition tests are going to be performed.~~ FSM state transitions still work and local runs can be taken (bitstream stored).

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: setup/scripts/xml/latency_scan_external_triggers.xml [hunk 1]
```diff
@@ -148,18 +148,6 @@
     <xc:Application class="gem::supervisor::GEMSupervisor" id="254" instance="0" network="local">
```
Comment (jsturdy) [inline setup/scripts/xml/latency_scan_external_triggers.xml:148]: @cgalloni should probably clean these up further in the calibration suite PR (whether complete removal, condense into one dummy config with configuration options described in the comments, etc.,)

File: setup/scripts/xml/latency_scan_lemo_triggers.xml [hunk 2]
```diff
@@ -148,18 +148,6 @@
     <xc:Application class="gem::supervisor::GEMSupervisor" id="254" instance="0" network="local">
```
Comment (jsturdy) [inline setup/scripts/xml/latency_scan_lemo_triggers.xml:148]: @cgalloni should probably clean these up further in the calibration suite PR

File: setup/scripts/xml/latency_scan_local_cal_pulse.xml [hunk 3]
```diff
@@ -147,18 +147,6 @@
     <xc:Application class="gem::supervisor::GEMSupervisor" id="254" instance="0" network="local">
```
Comment (jsturdy) [inline setup/scripts/xml/latency_scan_local_cal_pulse.xml:147]: @cgalloni should probably clean these up further in the calibration suite PR

File: setup/scripts/xml/threshold_scan_external_triggers.xml [hunk 4]
```diff
@@ -148,18 +148,6 @@
     <xc:Application class="gem::supervisor::GEMSupervisor" id="254" instance="0" network="local">
```
Comment (jsturdy) [inline setup/scripts/xml/threshold_scan_external_triggers.xml:148]: @cgalloni should probably clean these up further in the calibration suite PR

File: setup/scripts/xml/threshold_scan_local_triggers.xml [hunk 5]
```diff
@@ -148,18 +148,6 @@
     <xc:Application class="gem::supervisor::GEMSupervisor" id="254" instance="0" network="local">
       <properties xmlns="urn:xdaq-application:GEMSupervisor"
```
Comment (jsturdy) [inline setup/scripts/xml/threshold_scan_local_triggers.xml:149]: @cgalloni should probably clean these up further in the calibration suite PR

**Review by jsturdy — COMMENTED** [review]
needs to be rebased

File: setup/scripts/xml/latency_scan_external_triggers.xml (hunk 1 above)
Comment (lpetre-ulb) [reply setup/scripts/xml/latency_scan_external_triggers.xml:148]: :+1:

**Comment by lpetre-ulb** [comment]
Rebased!

**Review by mexanick — APPROVED** [review]

**Pull request merged**